docs/guides/running_service_different_networks.md

```diff
@@ -1,2 +1,114 @@
-!!! info
-    This tutorial will be added soon.
+!!! warning
+       <span style="color:red">**This section is currently being revised and restructured.
+       The current contents might contain redundant, inaccurate, or misplaced information.**<span style="color:blue">
+
+
+
+
+Sometimes agents and tests need to be run on specific networks. In the case of Ethereum,
+you might want to use developer environments like Hardhat or Ganache as your testbed,
+but at some point you will also require running on top of a public testnet like Ropsten or,
+ultimately, the main chain.
+
+To switch chains, you will first need to edit your chain configuration, as well as verify that
+the relevant addresses that come into play are funded. After that, the `valory/ledger` connection's
+configuration must be modified, and the process is slightly different for agents than for tests.
+Last but not least, it is important to keep in mind what the average block times are in the deploying network.
+
+## Ledger configuration for Agents
+Look at you agent's `aea-config.yaml` last section. It should look like this:
+```
+public_id: valory/ledger:0.19.0
+type: connection
+config:
+  ledger_apis:
+    ethereum:
+      address: ${LEDGER_ADDRESS:str:http://hardhat:8545}
+      chain_id: ${LEDGER_CHAIN_ID:int:31337}
+```
+There, edit the `address` and `chain_id` fields to match your requirements. In this case, we've set
+a local Hardhat node as our target chain, with its default id.
+
+## Ledger configuration for contract integration tests
```

Comment:
I am not sure there's much point to this guide on testing here. We should have a dedicated guide for testing elsewhere.